From 2bfbfdacf859ffc7a775fa75fcdf7032ea0eda02 Mon Sep 17 00:00:00 2001
From: Shaun Tancheff <shaun@tancheff.com>
Date: Tue, 15 Mar 2016 18:20:05 -0500
Subject: [PATCH 51/66] Add stream ID support for buffered
 mpage/__block_write_full_page()

Pass on the inode stream ID to the bio allocation.

Signed-off-by: Jens Axboe <axboe@fb.com>

Backported to v4.2
Signed-off-by: Shaun Tancheff <shaun.tancheff@seagate.com>
Signed-off-by: Shaun Tancheff <shaun@tancheff.com>
---
 fs/buffer.c | 8 ++++++--
 fs/mpage.c  | 1 +
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/fs/buffer.c b/fs/buffer.c
index 1cf7a53..0751326 100644
--- a/fs/buffer.c
+++ b/fs/buffer.c
@@ -1797,7 +1797,9 @@ static int __block_write_full_page(struct inode *inode, struct page *page,
 	do {
 		struct buffer_head *next = bh->b_this_page;
 		if (buffer_async_write(bh)) {
-			submit_bh_wbc(write_op, bh, 0, wbc);
+			submit_bh_wbc(write_op, bh,
+				      streamid_to_flags(inode_streamid(inode)),
+				      wbc);
 			nr_underway++;
 		}
 		bh = next;
@@ -1851,7 +1853,9 @@ recover:
 		struct buffer_head *next = bh->b_this_page;
 		if (buffer_async_write(bh)) {
 			clear_buffer_dirty(bh);
-			submit_bh_wbc(write_op, bh, 0, wbc);
+			submit_bh_wbc(write_op, bh,
+				      streamid_to_flags(inode_streamid(inode)),
+				      wbc);
 			nr_underway++;
 		}
 		bh = next;
diff --git a/fs/mpage.c b/fs/mpage.c
index ca0244b..be060ee 100644
--- a/fs/mpage.c
+++ b/fs/mpage.c
@@ -281,6 +281,7 @@ alloc_new:
 				GFP_KERNEL);
 		if (bio == NULL)
 			goto confused;
+		bio_set_streamid(bio, inode_streamid(inode));
 	}
 
 	length = first_hole << blkbits;
-- 
2.8.1

